src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -49,194 +47,96 @@ internal enum NullableFlowState : byte
         public TypeWithState WithNotNullState() => new TypeWithState(Type, NullableFlowState.NotNull);
         public TypeSymbolWithAnnotations ToTypeSymbolWithAnnotations()
         {
-            NullableAnnotation annotation = (this.State == NullableFlowState.NotNull)
-                ? NullableAnnotation.NotNullable : NullableAnnotation.Nullable;
+            NullableAnnotation annotation = this.State.IsNotNull() || Type?.CanContainNull() == false || Type?.IsTypeParameterDisallowingAnnotation() == true
+                ? NullableAnnotation.NotAnnotated : NullableAnnotation.Annotated;
             return TypeSymbolWithAnnotations.Create(this.Type, annotation);
         }
     }
 
     internal static class NullableAnnotationExtensions
     {
-        public static bool IsAnyNullable(this NullableAnnotation annotation)
+        public static bool IsAnnotated(this NullableAnnotation annotation)
         {
-            return annotation == NullableAnnotation.Annotated || annotation == NullableAnnotation.Nullable;
+            return annotation == NullableAnnotation.Annotated;
         }
 
-        public static bool IsAnyNotNullable(this NullableAnnotation annotation)
+        public static bool IsNotAnnotated(this NullableAnnotation annotation)
         {
-            return annotation == NullableAnnotation.NotAnnotated || annotation == NullableAnnotation.NotNullable;
+            return annotation == NullableAnnotation.NotAnnotated;
         }
 
-        public static bool IsSpeakable(this NullableAnnotation annotation)
+        public static bool IsOblivious(this NullableAnnotation annotation)
         {
-            return annotation == NullableAnnotation.Unknown ||
-                annotation == NullableAnnotation.NotAnnotated ||
-                annotation == NullableAnnotation.Annotated;
+            return annotation == NullableAnnotation.Unknown;
         }
 
-        /// <summary>
-        /// This method projects nullable annotations onto a smaller set that can be expressed in source.
-        /// </summary>
-        public static NullableAnnotation AsSpeakable(this NullableAnnotation annotation, TypeSymbol type)
+        public static bool MaybeNull(this NullableFlowState state)
```

Comment:
>public static bool MaybeNull(this NullableFlowState state) [](start = 8, length = 58)

nit: this extension should be in `NullableFlowStateExtensions` not `NullableAnnotationExtensions`.